src/events/bounded-event-queue.ts

```diff
@@ -1,20 +1,36 @@
 import { logger } from '../application-logger';
 import { MAX_EVENT_QUEUE_SIZE } from '../constants';
 
+import ArrayBackedNamedEventQueue from './array-backed-named-event-queue';
 import NamedEventQueue from './named-event-queue';
 
 /** A bounded event queue that drops events when it reaches its maximum size. */
-export class BoundedEventQueue<T> {
+export class BoundedEventQueue<T> implements NamedEventQueue<T> {
   constructor(
-    private readonly queue: NamedEventQueue<T>,
+    readonly name: string,
+    private readonly queue = new Array<T>(),
     private readonly maxSize = MAX_EVENT_QUEUE_SIZE,
   ) {}
 
+  length = this.queue.length;
+
+  splice(count: number): T[] {
+    return this.queue.splice(count);
+  }
+
+  isEmpty(): boolean {
+    return this.queue.length === 0;
+  }
+
+  [Symbol.iterator](): IterableIterator<T> {
```

Comment:
Your ruby is bleeding into typescript 😛

Reply:
lol this is actually part of the standard typescript API :) maybe they were inspired by ruby 🤣